### PR TITLE
fix: allow MQTT `encryption_enabled` with `json_enabled`

### PR DESCRIPTION
**Pull request by andrekir**
this change decouples JSON publishing from MeshPacket encryption and should allow MQTT using both `encryption_enabled` and `json_enabled` simultaneously.

### CODE REVIEW

**Review by caveman99 — APPROVED** [review]

**Review by thebentern — APPROVED** [review]

File: src/mesh/Router.cpp [hunk 1]
```diff
@@ -248,15 +248,7 @@ ErrorCode Router::send(meshtastic_MeshPacket *p)
...
-
-        if (moduleConfig.mqtt.enabled) {
-
-            LOG_INFO("Should encrypt MQTT?: %d\n", moduleConfig.mqtt.encryption_enabled);
-
-            // the packet is currently in a decrypted state.  send it now if they want decrypted packets
-            if (mqtt && !moduleConfig.mqtt.encryption_enabled)
-                mqtt->onSend(*p, chIndex);
-        }
+        meshtastic_MeshPacket *p_decoded = p;
```
Comment (GUVWAF) [inline src/mesh/Router.cpp:251]: Hmm, you're letting `p_decoded` point to the same memory location as `p` here. When `perhapsEncode()` modifies the memory pointed to by `p`, it also modifies where `p_decoded` points to. 

We'd need to make a copy, but then we also need to be careful that the memory gets released afterwards.

**Review by GUVWAF — CHANGES_REQUESTED** [review]

File: src/mesh/Router.cpp (hunk 1 above)
Comment (andrekir) [reply src/mesh/Router.cpp:251]: thanks for catching that. fix incoming.

File: src/mesh/Router.cpp [hunk 2]
```diff
@@ -248,15 +248,7 @@ ErrorCode Router::send(meshtastic_MeshPacket *p)
...
-
-        if (moduleConfig.mqtt.enabled) {
-
-            LOG_INFO("Should encrypt MQTT?: %d\n", moduleConfig.mqtt.encryption_enabled);
-
-            // the packet is currently in a decrypted state.  send it now if they want decrypted packets
-            if (mqtt && !moduleConfig.mqtt.encryption_enabled)
-                mqtt->onSend(*p, chIndex);
-        }
+        meshtastic_MeshPacket *p_decoded = new meshtastic_MeshPacket(*p);
```
Comment (GUVWAF) [inline src/mesh/Router.cpp:251]: For MeshPackets, in the rest of the code we use the `packetPool`. E.g. `packetPool.allocCopy()` and `packetPool.release()` for freeing.

File: src/mesh/Router.cpp [hunk 3]
```diff
@@ -248,15 +248,7 @@ ErrorCode Router::send(meshtastic_MeshPacket *p)
...
 
         auto encodeResult = perhapsEncode(p);
         if (encodeResult != meshtastic_Routing_Error_NONE) {
```
Comment (GUVWAF) [inline src/mesh/Router.cpp:254]: The copy should be freed before `return` here as well.

**Review by GUVWAF — CHANGES_REQUESTED** [review]

**Review by GUVWAF — APPROVED** [review]
Looks good now, thanks!

**Pull request merged**